File: topics_flutter/packages/domain/lib/topics/topic.dart
import 'package:flutter/foundation.dart';

@immutable
class Topic {
  final int id;
  final String title;
  final DateTime dateTime;
  final String description;

  const Topic({
    required this.id,
    required this.title,
    required this.dateTime,
    required this.description,
  });
}

File: topics_swiftui/topics/packages/domain/Sources/entities/Topic.swift
//
//  Topic.swift
//
//
//  Created by Marco Sanna on 29.04.24.
//

import Foundation

/// Holds information on a topic, like a title and the date
/// the topic will be presented
public struct Topic: Identifiable {
    public let id: String
    public let title: String
    public let description: String
    public let date: Date

    /// Public identifier needed to make it instantiable from other packages
    public init(id: String, title: String, description: String, date: Date) {
        self.id = id
        self.title = title
        self.description = description
        self.date = date
    }
}
